src/polkit/polkitunixprocess.c

```diff
@@ -1065,7 +1065,7 @@ polkit_unix_process_hash (PolkitSubject *subject)
 {
   PolkitUnixProcess *process = POLKIT_UNIX_PROCESS (subject);
 
-  return g_direct_hash (GSIZE_TO_POINTER ((polkit_unix_process_get_pid(process) + process->start_time))) ;
+  return g_direct_hash (GSIZE_TO_POINTER ((process->pid + process->start_time))) ;
```

Comment:
This is very wrong, the pid _must_ be resolved, as the entire point is to check if it did change, and if it did, it's no longer the same

Reply:
The hash doesn't need to be unique - simply unique enough to not cause performance degradation. The hash value must be constant throughout the object lifecycle for https://github.com/polkit-org/polkit/blob/894e11120d096b469677830aa94d40e309c22fe6/src/polkitbackend/polkitbackendinteractiveauthority.c#L2841 to work.

Reply:
No, this introduces a security risk. The pid must be resolved. Please drop this commit.

Reply:
The hash being used here is for a hash table. There should be no way for this to create a security issue.

Reply:
Note that it's assumed here that the `pid` field will be 0 if `pidfd` is set, which means `start_time` alone will be used for hashing.